besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1502,10 +1505,46 @@ private BesuCommand configure() throws Exception {
         .ifPresent(p -> ensureAllNodesAreInAllowlist(staticNodes, p));
     metricsConfiguration = metricsConfiguration();
 
+    if (isGoQuorumCompatibilityMode) {
+      configureGoQuorumPrivacy();
+    }
+
     logger.info("Security Module: {}", securityModuleName);
     return this;
   }
 
+  private void configureGoQuorumPrivacy() {
+
+    GoQuorumPrivacyParameters.isEnabled = true;
+    final EnclaveFactory enclaveFactory = new EnclaveFactory(Vertx.vertx());
+    if (privacyKeyStoreFile != null) {
+      GoQuorumPrivacyParameters.goQuorumEnclave =
+          enclaveFactory.createGoQuorumEnclave(
+              privacyUrl,
+              privacyKeyStoreFile,
+              privacyKeyStorePasswordFile,
+              privacyTlsKnownEnclaveFile);
+    } else {
+      GoQuorumPrivacyParameters.goQuorumEnclave = enclaveFactory.createGoQuorumEnclave(privacyUrl);
+    }
```

Comment:
NIT: it would be nice to see this logic into its own method. Something like `createGoQuorumEnclave` or similar.